Subject: [PATCH] feature: 增加dolphinscheduler 3.2.2
---
Index: bigtop-packages/src/common/dolphinscheduler/install_dolphinscheduler.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop-packages/src/common/dolphinscheduler/install_dolphinscheduler.sh b/bigtop-packages/src/common/dolphinscheduler/install_dolphinscheduler.sh
new file mode 100644
--- /dev/null	(date 1728716787165)
+++ b/bigtop-packages/src/common/dolphinscheduler/install_dolphinscheduler.sh	(date 1728716787165)
@@ -0,0 +1,172 @@
+#!/bin/bash
+
+set -ex
+
+usage() {
+  cat <<EOF
+Usage: $0 <options>
+  Required options:
+     --build-dir=DIR             Path to build directory for DolphinScheduler
+     --prefix=PREFIX             Path to install into
+     --tar-name=NAME             Tarball name for extraction
+
+  Optional options:
+     --bin-dir=DIR               Path to install binaries [/usr/lib/dolphinscheduler/bin]
+     --conf-dir=DIR              Path to configuration files provided by the package [/etc/dolphinscheduler/conf.dist]
+     --doc-dir=DIR               Path to install documentation [/usr/share/doc/dolphinscheduler]
+     --etc-dir=DIR               Path to install configuration files [/etc/dolphinscheduler]
+     --examples-dir=DIR          Path to install examples [doc-dir/examples]
+     --installed-lib-dir=DIR     Path where lib-dir will end up on target system
+     --lib-dir=DIR               Path to install dolphinscheduler home [/usr/lib/dolphinscheduler]
+     --man-dir=DIR               Path to install man pages [/usr/share/man/man1]
+     --var-dir=DIR               Path to install variable data
+EOF
+  exit 1
+}
+
+OPTS=$(getopt \
+  -n $0 \
+  -o '' \
+  -l 'bin-dir:' \
+  -l 'build-dir:' \
+  -l 'conf-dir:' \
+  -l 'doc-dir:' \
+  -l 'etc-dir:' \
+  -l 'examples-dir:' \
+  -l 'installed-lib-dir:' \
+  -l 'lib-dir:' \
+  -l 'man-dir:' \
+  -l 'prefix:' \
+  -l 'tar-name:' \
+  -l 'var-dir:' -- "$@")
+
+if [ $? != 0 ]; then
+  usage
+fi
+
+eval set -- "$OPTS"
+
+PREFIX=""
+BUILD_DIR=""
+TAR_NAME=""
+
+while true; do
+  case "$1" in
+  --bin-dir)
+    BIN_DIR=$2
+    shift 2
+    ;;
+  --build-dir)
+    BUILD_DIR=$2
+    shift 2
+    ;;
+  --conf-dir)
+    CONF_DIR=$2
+    shift 2
+    ;;
+  --doc-dir)
+    DOC_DIR=$2
+    shift 2
+    ;;
+  --etc-dir)
+    ETC_DIR=$2
+    shift 2
+    ;;
+  --installed-lib-dir)
+    INSTALLED_LIB_DIR=$2
+    shift 2
+    ;;
+  --lib-dir)
+    LIB_DIR=$2
+    shift 2
+    ;;
+  --man-dir)
+    MAN_DIR=$2
+    shift 2
+    ;;
+  --prefix)
+    PREFIX=$2
+    shift 2
+    ;;
+  --tar-name)
+    TAR_NAME=$2
+    shift 2
+    ;;
+  --var-dir)
+    VAR_DIR=$2
+    shift 2
+    ;;
+  --)
+    shift
+    break
+    ;;
+  *)
+    echo "Unknown option: $1"
+    usage
+    exit 1
+    ;;
+  esac
+done
+
+if [ -z "$PREFIX" ] || [ -z "$BUILD_DIR" ] || [ -z "$TAR_NAME" ]; then
+  echo "Missing required parameters."
+  usage
+fi
+
+# Set default directories if not set
+DOC_DIR=${DOC_DIR:-/usr/share/doc/dolphinscheduler}
+LIB_DIR=${LIB_DIR:-/usr/lib/dolphinscheduler}
+BIN_DIR=${BIN_DIR:-/usr/lib/dolphinscheduler/bin}
+ETC_DIR=${ETC_DIR:-/etc/dolphinscheduler}
+MAN_DIR=${MAN_DIR:-/usr/share/man/man1}
+VAR_DIR=${VAR_DIR:-/var/lib/dolphinscheduler}
+CONF_DIR=${CONF_DIR:-""}
+
+ALERT_CONF_DIR=${ETC_DIR}/alert-server/conf.empty
+API_CONF_DIR=${ETC_DIR}/api-server/conf.empty
+MASTER_CONF_DIR=${ETC_DIR}/master-server/conf.empty
+WORKER_CONF_DIR=${ETC_DIR}/worker-server/conf.empty
+
+# Log directories
+NP_LOG_DIR=/var/log/dolphinscheduler
+NP_ALERT_LOG_DIR=${NP_LOG_DIR}/alert-server
+NP_API_LOG_DIR=${NP_LOG_DIR}/api-server
+NP_MASTER_LOG_DIR=${NP_LOG_DIR}/master-server
+NP_WORKER_LOG_DIR=${NP_LOG_DIR}/worker-server
+
+# Configuration directories
+NP_ETC_DIR=/etc/dolphinscheduler
+#NP_ALERT_CONF_DIR=${NP_ETC_DIR}/alert-server
+#NP_API_CONF_DIR=${NP_ETC_DIR}/api-server
+#NP_MASTER_CONF_DIR=${NP_ETC_DIR}/master-server
+#NP_WORKER_CONF_DIR=${NP_ETC_DIR}/worker-server
+
+# Create required directories
+for dir in \
+  "${PREFIX}${NP_ALERT_LOG_DIR}" "${PREFIX}${NP_API_LOG_DIR}" "${PREFIX}${NP_MASTER_LOG_DIR}" "${PREFIX}${NP_WORKER_LOG_DIR}" \
+  "${PREFIX}${MAN_DIR}" "${PREFIX}${BIN_DIR}" "${PREFIX}${DOC_DIR}" "${PREFIX}${LIB_DIR}" \
+  "${PREFIX}${CONF_DIR}" "${PREFIX}${ETC_DIR}" "${PREFIX}${VAR_DIR}" \
+  "${PREFIX}${ALERT_CONF_DIR}" "${PREFIX}${API_CONF_DIR}" "${PREFIX}${MASTER_CONF_DIR}" "${PREFIX}${WORKER_CONF_DIR}"; do
+  install -d -m 0755 "$dir"
+done
+
+# Extract tarball directly to LIB_DIR
+tar -zxvf ${BUILD_DIR}/${TAR_NAME}/dolphinscheduler-dist/target/apache-${TAR_NAME}-bin.tar.gz -C ${PREFIX}${LIB_DIR} --strip-components=1
+
+# Move configuration files to their respective directories
+for svc in alert-server api-server master-server worker-server; do
+  mkdir -p ${PREFIX}/etc/dolphinscheduler/$svc/
+  mv ${PREFIX}${LIB_DIR}/$svc/conf/* ${PREFIX}/etc/dolphinscheduler/$svc/
+  rm -rf ${PREFIX}${LIB_DIR}/$svc/conf
+  # 创建符号链接，确保它指向最终安装路径
+  ln -s ${NP_ETC_DIR}/$svc/conf ${PREFIX}${LIB_DIR}/$svc
+  # 日志目录的符号链接
+  ln -s ${NP_LOG_DIR}/$svc ${PREFIX}${LIB_DIR}/$svc/logs
+done
+
+# Set permissions for JAR files
+for svc in alert-server api-server master-server worker-server; do
+  chmod -R 644 ${PREFIX}${LIB_DIR}/$svc/libs/*
+done
+
+echo "Installation completed successfully."
Index: bigtop-packages/src/common/phoenix/install_phoenix.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop-packages/src/common/phoenix/install_phoenix.sh b/bigtop-packages/src/common/phoenix/install_phoenix.sh
--- a/bigtop-packages/src/common/phoenix/install_phoenix.sh	(revision cd9e319a34af99612b8da1fdaad812763e3c857f)
+++ b/bigtop-packages/src/common/phoenix/install_phoenix.sh	(date 1728716988442)
@@ -39,12 +39,16 @@
   -n $0 \
   -o '' \
   -l 'prefix:' \
+  -l 'bin-dir:' \
+  -l 'conf-dir:' \
   -l 'doc-dir:' \
+  -l 'etc-dir:' \
+  -l 'examples-dir:' \
   -l 'lib-dir:' \
+  -l 'man-dir:' \
+  -l 'tar-name:' \
+  -l 'var-dir:' \
   -l 'installed-lib-dir:' \
-  -l 'bin-dir:' \
-  -l 'examples-dir:' \
-  -l 'conf-dir:' \
   -l 'build-dir:' -- "$@")

 if [ $? != 0 ] ; then
@@ -52,6 +56,7 @@
 fi

 eval set -- "$OPTS"
+set -ex
 while true ; do
     case "$1" in
         --prefix)
@@ -60,24 +65,39 @@
         --build-dir)
         BUILD_DIR=$2 ; shift 2
         ;;
+        --bin-dir)
+        BIN_DIR=$2 ; shift 2
+        ;;
+        --conf-dir)
+        CONF_DIR=$2 ; shift 2
+        ;;
         --doc-dir)
         DOC_DIR=$2 ; shift 2
         ;;
+        --etc-dir)
+        ETC_DIR=$2 ; shift 2
+        ;;
         --lib-dir)
         LIB_DIR=$2 ; shift 2
         ;;
+        --man-dir)
+        MAN_DIR=$2 ; shift 2
+        ;;
+        --tar-name)
+        TAR_NAME=$2 ; shift 2
+        ;;
+        --tar-name)
+        TAR_NAME=$2 ; shift 2
+        ;;
+        --var-dir)
+        VAR_DIR=$2 ; shift 2
+        ;;
         --installed-lib-dir)
         INSTALLED_LIB_DIR=$2 ; shift 2
         ;;
-        --bin-dir)
-        BIN_DIR=$2 ; shift 2
-        ;;
         --examples-dir)
         EXAMPLES_DIR=$2 ; shift 2
         ;;
-        --conf-dir)
-        CONF_DIR=$2 ; shift 2
-        ;;
         --)
         shift ; break
         ;;
@@ -113,12 +133,12 @@
 install -d -m 0755 $PREFIX/var/lib/phoenix
 install -d -m 0755 $PREFIX/var/log/phoenix

-cp $BUILD_DIR/*.jar $PREFIX/$LIB_DIR/
-cp -r $BUILD_DIR/bin $PREFIX/$LIB_DIR/
+cp $BUILD_DIR/$TAR_NAME/build/*.jar $PREFIX/$LIB_DIR/
+cp -r $BUILD_DIR/$TAR_NAME/bin $PREFIX/$LIB_DIR/
 chmod 755 $PREFIX/$BIN_DIR/*.py

-cp -a $BUILD_DIR/{LICENSE,NOTICE} $PREFIX/$DOC_DIR/
-cp -ra $BUILD_DIR/examples $PREFIX/$LIB_DIR/
+cp -a $BUILD_DIR/$TAR_NAME/{LICENSE,NOTICE} $PREFIX/$DOC_DIR/
+cp -ra $BUILD_DIR/$TAR_NAME/examples $PREFIX/$LIB_DIR/

 # Remove the executable bit from jars to avoid lintian warnings
 find $PREFIX/$LIB_DIR -name '*.jar' -exec chmod a-x {} \;
Index: bigtop.bom
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop.bom b/bigtop.bom
--- a/bigtop.bom	(revision cd9e319a34af99612b8da1fdaad812763e3c857f)
+++ b/bigtop.bom	(date 1728716938298)
@@ -283,6 +283,7 @@

         'phoenix' {
             name = 'phoenix'
+            rpm_pkg_suffix = "_" + bigtop.base_version.replace(".", "_")
             relNotes = 'Apache Phoenix: A SQL skin over HBase'
             version { base = "5.1.2"; pkg = base; release = 1 }
             tarball {
@@ -294,6 +295,7 @@
                 site = "${apache.APACHE_MIRROR}/${download_path}"
                 archive = "${apache.APACHE_ARCHIVE}/${download_path}"
             }
+            maven_parallel_build = true
         }

         'bigtop-groovy' {
@@ -506,5 +508,20 @@
                 archive = site
             }
         }
+
+        'dolphinscheduler' {
+            name = 'dolphinscheduler'
+            rpm_pkg_suffix = "_" + bigtop.base_version.replace(".", "_")
+            relNotes = 'Dolphinscheduler v3.2.2 Release'
+            version { base = '3.2.2'; pkg = base; release = 1 }
+            tarball {
+                destination = "${name}-${version.base}.tar.gz"
+                source = "${version.base}.tar.gz"
+            }
+            url {
+                site = "https://ghgo.xyz/https://github.com/apache/dolphinscheduler/archive/refs/tags"
+                archive = site
+            }
+        }
     }
 }
Index: bigtop-packages/src/common/dolphinscheduler/do-component-build
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop-packages/src/common/dolphinscheduler/do-component-build b/bigtop-packages/src/common/dolphinscheduler/do-component-build
new file mode 100644
--- /dev/null	(date 1728694626731)
+++ b/bigtop-packages/src/common/dolphinscheduler/do-component-build	(date 1728694626731)
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# 开启错误处理和详细模式
+set -ex
+
+# 获取当前脚本所在目录，并引入 bigtop.bom 文件
+SCRIPT_DIR=$(dirname "$0")
+source "$SCRIPT_DIR/bigtop.bom"
+
+# Maven 编译选项配置
+DOLPHIN_MAVEN_OPTS=(
+  "-Prelease"
+  "-Dmaven.test.skip=true"
+  "-Dhadoop.version=${HADOOP_VERSION}"
+  "-Dcurator.version=4.2.0"
+  "-Dzookeeper.version=${ZOOKEEPER_VERSION}"
+  "-Dspark.version=${SPARK_VERSION}"
+  "-Dhive.version=${HIVE_VERSION}"
+)
+
+# 修正格式
+mvn spotless:apply
+# 将数组展开成字符串，并传递给 mvn 命令
+mvn -X clean install "${DOLPHIN_MAVEN_OPTS[@]}" "$@"
Index: bigtop-packages/src/rpm/dolphinscheduler/SPECS/dolphinscheduler.spec
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop-packages/src/rpm/dolphinscheduler/SPECS/dolphinscheduler.spec b/bigtop-packages/src/rpm/dolphinscheduler/SPECS/dolphinscheduler.spec
new file mode 100644
--- /dev/null	(date 1728716787160)
+++ b/bigtop-packages/src/rpm/dolphinscheduler/SPECS/dolphinscheduler.spec	(date 1728716787160)
@@ -0,0 +1,137 @@
+# 定义组件名称
+%define component_name dolphinscheduler
+%define group_name hadoop
+%define user_name %{component_name}
+
+# 定义组件描述内容
+%define component_summary Apache DolphinScheduler is the modern data workflow orchestration platform.
+%define component_license ASL 2.0
+%define component_url https://dolphinscheduler.apache.org/
+%define component_base_version %{dolphinscheduler_base_version}
+%define component_version %{dolphinscheduler_version}
+%define component_release %{dolphinscheduler_release}
+
+#定义 dl 解压前压缩包的名称
+%define component_tarball_origin_name %{component_name}-%{component_base_version}.tar.gz
+# 定义 dl 解压后的源代码包名称
+%define component_tarball_name %{component_name}-%{component_base_version}
+
+
+# 禁用调试信息包
+%define debug_package %{nil}
+# 禁用符号链接检查
+%define __os_install_post %{nil}
+%define __jar_repack %{nil}
+# 定义自动切换配置命令
+%define component_pkg_name %{component_name}%{pkg_name_suffix}
+%define alternatives_cmd alternatives
+%define etc_default %{parent_dir}/etc/default
+
+# 定义 usr、var、etc 三大目录的路径
+%define usr_lib_component %{parent_dir}/usr/lib/%{component_name}
+%define var_lib_component %{parent_dir}/var/lib/%{component_name}
+%define etc_component %{parent_dir}/etc/%{component_name}
+
+# 定义 bin、man、doc 三大目录的路径
+%define bin_dir %{parent_dir}%{_bindir}
+%define man_dir %{parent_dir}%{_mandir}
+%define doc_dir %{parent_dir}%{_docdir}
+
+# 定义无前缀的运行路径、日志路径和配置路径
+%define np_var_log_component /var/log/%{component_name}
+%define np_var_run_component /var/run/%{component_name}
+%define np_etc_component /etc/%{component_name}
+
+%define component_services alert-server api-server master-server worker-server
+
+Name: %{component_pkg_name}
+Version: %{component_version}
+Release: %{component_release}
+Summary: %{component_summary}
+License: %{component_license}
+URL: %{component_url}
+Group: Development/Libraries
+Buildroot: %{_topdir}/INSTALL/%{name}-%{version}
+Source0: %{component_tarball_origin_name}
+Source1: do-component-build
+Source2: install_%{component_name}.sh
+#BIGTOP_PATCH_FILES
+Requires: psmisc, sudo, coreutils, /usr/sbin/useradd, /usr/sbin/usermod, /sbin/chkconfig, /sbin/service, /usr/bin/lsb_release, /bin/sh
+Requires: bigtop-select >= 3.2.0
+
+%description
+Apache DolphinScheduler is the modern data workflow orchestration platform.
+
+%prep
+# 解压缩源码包
+%setup -n %{component_tarball_name}
+
+#BIGTOP_PATCH_COMMANDS
+
+%build
+env COMPONENT_VERSION=%{version} bash %{SOURCE1}
+
+%install
+%__rm -rf $RPM_BUILD_ROOT
+bash %{SOURCE2} \
+    --build-dir=%{_topdir}/BUILD \
+    --man-dir=%{man_dir} \
+    --bin-dir=%{bin_dir} \
+    --doc-dir=%{doc_dir} \
+    --lib-dir=%{usr_lib_component} \
+    --var-dir=%{var_lib_component} \
+    --etc-dir=%{etc_component} \
+    --tar-name=%{component_tarball_name} \
+    --prefix=$RPM_BUILD_ROOT
+
+%clean
+rm -rf $RPM_BUILD_ROOT
+
+%pre
+if [ "$1" = 1 ]; then
+    getent group %{group_name} >/dev/null  || groupadd -r %{group_name}
+    getent passwd %{user_name} >/dev/null || /usr/sbin/useradd --comment "%{user_name}" --shell /bin/bash -M -r -g %{group_name} -G hadoop --home /home/%{user_name} %{user_name}
+fi
+
+%post
+if [ "$1" = 1 ]; then
+  for service in %{component_services}; do
+     # 初始化 /etc/dolphinscheduler 组件目录
+     mkdir -p %{np_etc_component}/${service}
+     # 设置 etc 目录
+     %{alternatives_cmd} --install %{np_etc_component}/${service}/conf ${service}-conf %{etc_component}/${service}/conf.empty 30
+     # 设置日志目录
+     mkdir -p %{np_var_log_component}/${service}
+  done
+  mkdir -p /var/lib/%{component_name}
+  chown -R %{user_name}:%{group_name} /var/lib/%{component_name}
+fi
+
+%preun
+if [ "$1" = 0 ]; then
+  for service in %{component_services}; do
+     # 移除 etc 目录
+     %{alternatives_cmd} --remove ${service}-conf %{etc_component}/${service}/conf.empty || :
+  done
+fi
+
+%postun
+if [ "$1" = 0 ]; then
+  rm -rf %{np_var_log_component}/
+fi
+
+%files
+%defattr(-,%{user_name},%{group_name})
+%{usr_lib_component}
+%{var_lib_component}
+%{etc_component}
+%attr(0775,%{user_name},%{group_name}) %{np_var_log_component}
+# 包含具体的 /etc/dolphinscheduler 下的文件
+%config(noreplace) /etc/dolphinscheduler/alert-server/*
+%config(noreplace) /etc/dolphinscheduler/api-server/*
+%config(noreplace) /etc/dolphinscheduler/master-server/*
+%config(noreplace) /etc/dolphinscheduler/worker-server/*
+
+%changelog
+* Thu Oct 10 2024 JaneTTR <3832514048@qq.com> - 1.0-1
+- Initial package.
\ No newline at end of file
Index: bigtop-packages/src/common/dolphinscheduler/patch0-DATASOURCE-OPTIMIZED.diff
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop-packages/src/common/dolphinscheduler/patch0-DATASOURCE-OPTIMIZED.diff b/bigtop-packages/src/common/dolphinscheduler/patch0-DATASOURCE-OPTIMIZED.diff
new file mode 100644
--- /dev/null	(date 1728723923035)
+++ b/bigtop-packages/src/common/dolphinscheduler/patch0-DATASOURCE-OPTIMIZED.diff	(date 1728723923035)
@@ -0,0 +1,89 @@
+Subject: [PATCH] optimized: 增加mysql 驱动
+---
+Index: dolphinscheduler-worker/pom.xml
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+diff --git a/dolphinscheduler-worker/pom.xml b/dolphinscheduler-worker/pom.xml
+--- a/dolphinscheduler-worker/pom.xml	(revision 2ec95b5c2c50d2882d40e07fb13cf38b5f0469b4)
++++ b/dolphinscheduler-worker/pom.xml	(date 1728530771659)
+@@ -118,6 +118,13 @@
+             <artifactId>zt-zip</artifactId>
+         </dependency>
+
++        <dependency>
++            <groupId>mysql</groupId>
++            <artifactId>mysql-connector-java</artifactId>
++            <version>8.0.24</version>
++            <scope>compile</scope>
++        </dependency>
++
+     </dependencies>
+
+     <build>
+Index: dolphinscheduler-alert/dolphinscheduler-alert-server/pom.xml
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+diff --git a/dolphinscheduler-alert/dolphinscheduler-alert-server/pom.xml b/dolphinscheduler-alert/dolphinscheduler-alert-server/pom.xml
+--- a/dolphinscheduler-alert/dolphinscheduler-alert-server/pom.xml	(revision 2ec95b5c2c50d2882d40e07fb13cf38b5f0469b4)
++++ b/dolphinscheduler-alert/dolphinscheduler-alert-server/pom.xml	(date 1728530601997)
+@@ -71,6 +71,13 @@
+             <artifactId>spring-boot-starter-test</artifactId>
+             <scope>test</scope>
+         </dependency>
++
++        <dependency>
++            <groupId>mysql</groupId>
++            <artifactId>mysql-connector-java</artifactId>
++            <version>8.0.24</version>
++            <scope>compile</scope>
++        </dependency>
+     </dependencies>
+
+     <build>
+Index: dolphinscheduler-api/pom.xml
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+diff --git a/dolphinscheduler-api/pom.xml b/dolphinscheduler-api/pom.xml
+--- a/dolphinscheduler-api/pom.xml	(revision 2ec95b5c2c50d2882d40e07fb13cf38b5f0469b4)
++++ b/dolphinscheduler-api/pom.xml	(date 1728530716579)
+@@ -236,6 +236,12 @@
+             <groupId>org.apache.dolphinscheduler</groupId>
+             <artifactId>dolphinscheduler-extract-alert</artifactId>
+         </dependency>
++        <dependency>
++            <groupId>mysql</groupId>
++            <artifactId>mysql-connector-java</artifactId>
++            <version>8.0.24</version>
++            <scope>compile</scope>
++        </dependency>
+     </dependencies>
+     <build>
+         <testResources>
+Index: dolphinscheduler-master/pom.xml
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+diff --git a/dolphinscheduler-master/pom.xml b/dolphinscheduler-master/pom.xml
+--- a/dolphinscheduler-master/pom.xml	(revision 2ec95b5c2c50d2882d40e07fb13cf38b5f0469b4)
++++ b/dolphinscheduler-master/pom.xml	(date 1728530744894)
+@@ -289,6 +289,13 @@
+             <groupId>org.springframework.cloud</groupId>
+             <artifactId>spring-cloud-starter-kubernetes-client-config</artifactId>
+         </dependency>
++
++        <dependency>
++            <groupId>mysql</groupId>
++            <artifactId>mysql-connector-java</artifactId>
++            <version>8.0.24</version>
++            <scope>compile</scope>
++        </dependency>
+     </dependencies>
+
+     <build>
Index: bigtop-packages/src/rpm/phoenix/SPECS/phoenix.spec
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop-packages/src/rpm/phoenix/SPECS/phoenix.spec b/bigtop-packages/src/rpm/phoenix/SPECS/phoenix.spec
--- a/bigtop-packages/src/rpm/phoenix/SPECS/phoenix.spec	(revision cd9e319a34af99612b8da1fdaad812763e3c857f)
+++ b/bigtop-packages/src/rpm/phoenix/SPECS/phoenix.spec	(date 1728716961010)
@@ -12,20 +12,25 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-%define phoenix_home /usr/lib/%{name}
+%define phoenix_name phoenix
+%define phoenix_pkg_name phoenix%{pkg_name_suffix}
 %define bin_phoenix %{phoenix_home}/bin
 %define examples_phoenix %{phoenix_home}/examples
-%define etc_phoenix_conf %{_sysconfdir}/%{name}/conf
+%define man_dir %{parent_dir}/%{_mandir}
+%define zookeeper_home %{parent_dir}/usr/lib/zookeeper
+%define hadoop_home %{parent_dir}/usr/lib/hadoop
+%define hadoop_mapreduce_home %{parent_dir}/usr/lib/hadoop-mapreduce
+%define hadoop_yarn_home %{parent_dir}/usr/lib/hadoop-yarn
+%define hadoop_hdfs_home %{parent_dir}/usr/lib/hadoop-hdfs
+%define hbase_home %{parent_dir}/usr/lib/hbase
+%define tarball_name %{phoenix_name}-%{phoenix_base_version}
 %define etc_phoenix_conf_dist %{etc_phoenix_conf}.dist
-%define var_lib_phoenix /var/lib/%{name}
-%define var_log_phoenix /var/log/%{name}
-%define man_dir %{_mandir}
-%define zookeeper_home /usr/lib/zookeeper
-%define hadoop_home /usr/lib/hadoop
-%define hadoop_mapreduce_home /usr/lib/hadoop-mapreduce
-%define hadoop_yarn_home /usr/lib/hadoop-yarn
-%define hadoop_hdfs_home /usr/lib/hadoop-hdfs
-%define hbase_home /usr/lib/hbase
+%define prefix_var_lib_phoenix %{parent_dir}/var/lib/%{phoenix_name}
+%define prefix_var_log_phoenix %{parent_dir}/var/log/%{phoenix_name}
+%define etc_phoenix %{parent_dir}/%{_sysconfdir}/%{phoenix_name}
+%define etc_phoenix_conf %{etc_phoenix}/conf
+%define phoenix_home %{parent_dir}/usr/lib/%{phoenix_name}
+
 #BIGTOP_PATCH_FILES

 %if  %{?suse_version:1}0
@@ -45,7 +50,7 @@
     /usr/lib/rpm/brp-compress ; \
     %{nil}

-%define doc_phoenix %{_docdir}/%{name}
+%define doc_phoenix %{parent_dir}/%{_docdir}
 %define alternatives_cmd update-alternatives

 %else
@@ -67,12 +72,12 @@
     %{nil}
 %endif

-%define doc_phoenix %{_docdir}/%{name}-%{phoenix_version}
+%define doc_phoenix %{parent_dir}/%{_docdir}/%{phoenix_name}-%{phoenix_version}
 %define alternatives_cmd alternatives

 %endif

-Name: phoenix
+Name: %{phoenix_pkg_name}
 Version: %{phoenix_version}
 Release: %{phoenix_release}
 Summary: Phoenix is a SQL skin over HBase and client-embedded JDBC driver.
@@ -80,7 +85,7 @@
 Group: Development/Libraries
 Buildroot: %{_topdir}/INSTALL/%{name}-%{version}
 License: ASL 2.0
-Source0: %{name}-%{phoenix_base_version}-src.tar.gz
+Source0: %{phoenix_name}-%{phoenix_base_version}-src.tar.gz
 Source1: do-component-build
 Source2: install_phoenix.sh
 Source3: bigtop.bom
@@ -109,7 +114,7 @@
 standard JDBC interface; all the usual interfaces are supported.

 %prep
-%setup -n %{name}-%{phoenix_base_version}
+%setup -n %{tarball_name}
 #BIGTOP_PATCH_COMMANDS

 %build
@@ -118,16 +123,22 @@
 %install
 %__rm -rf $RPM_BUILD_ROOT
 bash %{SOURCE2} \
-  --build-dir=build \
-  --doc-dir=%{doc_phoenix} \
-  --prefix=$RPM_BUILD_ROOT
+    --build-dir=%{_topdir}/BUILD \
+    --bin-dir=%{bin_phoenix} \
+    --doc-dir=%{doc_phoenix} \
+    --etc-dir=%{etc_phoenix} \
+    --lib-dir=%{phoenix_home} \
+    --man-dir=%{man_dir} \
+    --tar-name=%{tarball_name} \
+    --var-dir=%{prefix_var_lib_phoenix} \
+    --prefix=$RPM_BUILD_ROOT

 %pre
 getent group phoenix >/dev/null || groupadd -r phoenix
-getent passwd phoenix >/dev/null || useradd -c "Phoenix" -s /sbin/nologin -g phoenix -r -d %{var_lib_phoenix} phoenix 2> /dev/null || :
+getent passwd phoenix >/dev/null || useradd -c "Phoenix" -s /sbin/nologin -g phoenix -r -d %{prefix_var_lib_phoenix} phoenix 2> /dev/null || :

 %post
-%{alternatives_cmd} --install %{etc_phoenix_conf} %{name}-conf %{etc_phoenix_conf_dist} 30
+%{alternatives_cmd} --install %{etc_phoenix_conf} %{phoenix_name}-conf %{etc_phoenix_conf_dist} 30

 %if  0%{?rhel} >= 8
 %{alternatives_cmd} --set python /usr/bin/python3
@@ -136,7 +147,7 @@

 %preun
 if [ "$1" = 0 ]; then
-  %{alternatives_cmd} --remove %{name}-conf %{etc_phoenix_conf_dist} || :
+  %{alternatives_cmd} --remove %{phoenix_name}-conf %{etc_phoenix_conf_dist} || :
 fi


