Subject: [PATCH] feature: 修复3.2.2 缺依赖问题
---
Index: bigtop-packages/src/common/dolphinscheduler/patch1-TOOLS-FIXED.diff
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop-packages/src/common/dolphinscheduler/patch1-TOOLS-FIXED.diff b/bigtop-packages/src/common/dolphinscheduler/patch1-TOOLS-FIXED.diff
new file mode 100644
--- /dev/null	(date 1728877840689)
+++ b/bigtop-packages/src/common/dolphinscheduler/patch1-TOOLS-FIXED.diff	(date 1728877840689)
@@ -0,0 +1,30 @@
+Subject: [PATCH] optimized: 增加mysql 驱动
+---
+Index: dolphinscheduler-tools/pom.xml
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+diff --git a/dolphinscheduler-tools/pom.xml b/dolphinscheduler-tools/pom.xml
+--- a/dolphinscheduler-tools/pom.xml	(revision 2ec95b5c2c50d2882d40e07fb13cf38b5f0469b4)
++++ b/dolphinscheduler-tools/pom.xml	(date 1728877566153)
+@@ -129,16 +129,19 @@
+         <dependency>
+             <groupId>org.testcontainers</groupId>
+             <artifactId>mysql</artifactId>
++            <scope>compile</scope>
+         </dependency>
+
+         <dependency>
+             <groupId>mysql</groupId>
+             <artifactId>mysql-connector-java</artifactId>
++            <scope>compile</scope>
+         </dependency>
+
+         <dependency>
+             <groupId>org.testcontainers</groupId>
+             <artifactId>postgresql</artifactId>
++            <scope>compile</scope>
+         </dependency>
+
+     </dependencies>
Index: bigtop-packages/src/common/dolphinscheduler/install_dolphinscheduler.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bigtop-packages/src/common/dolphinscheduler/install_dolphinscheduler.sh b/bigtop-packages/src/common/dolphinscheduler/install_dolphinscheduler.sh
--- a/bigtop-packages/src/common/dolphinscheduler/install_dolphinscheduler.sh	(revision b1b87d5e04b601e0bdf575538b3640c3f845266d)
+++ b/bigtop-packages/src/common/dolphinscheduler/install_dolphinscheduler.sh	(date 1728878025059)
@@ -155,8 +155,8 @@

 # Move configuration files to their respective directories
 for svc in alert-server api-server master-server worker-server; do
-  mkdir -p ${PREFIX}/etc/dolphinscheduler/$svc/
-  mv ${PREFIX}${LIB_DIR}/$svc/conf/* ${PREFIX}/etc/dolphinscheduler/$svc/
+  mkdir -p ${PREFIX}/etc/dolphinscheduler/$svc/conf
+  mv ${PREFIX}${LIB_DIR}/$svc/conf/* ${PREFIX}/etc/dolphinscheduler/$svc/conf
   rm -rf ${PREFIX}${LIB_DIR}/$svc/conf
   # 创建符号链接，确保它指向最终安装路径
   ln -s ${NP_ETC_DIR}/$svc/conf ${PREFIX}${LIB_DIR}/$svc
